### DISCUSSION TITLE
saltstack issues (after hotfixes 20220401 & 20220405)

### DISCUSSION THREAD

**Discussion by 73F0SG1R** [Security-Onion-Solutions/securityonion#7769] [Unsupported Versions]
network (ubuntu 20.04) install

sudo soup 
...
Checking to see if there are hotfixes needed
You are already running the latest version of Security Onion.
root@somgr:/home/xxxx#

trying to add search node:

Results: The key glob 'sosearch_searchnode' does not match any unaccepted keys.
The key glob 'sosearch_searchnode' does not match any unaccepted keys. (0)
Did not find expectedOutput: 'going to be accepted' in the output below from running the command: 'accept_salt_key_remote'
<Start of output>
The key glob 'sosearch_searchnode' does not match any unaccepted keys.
The key glob 'sosearch_searchnode' does not match any unaccepted keys.
<End of output>
Command failed with exit code 0; will retry in 10 seconds (20 / 20)...
Command continues to fail; giving up.
...
Adding sosearch_searchnode
Syncing all salt modules.
Attempt to authenticate with the salt master failed with timeout error
...
Errors detected during setup; skipping post-setup steps to allow for analysis of failures.
[sosetup.log](https://github.com/Security-Onion-Solutions/securityonion/files/8459586/sosetup.log)

please see attached sosetup.log

**Comment by dougburks** [comment]
Have you tried upgrading your manager to the latest 20220407 hotfix?
https://blog.securityonion.net/2022/04/security-onion-23110-20220407-hotfix.html

**Reply by 73F0SG1R** [reply]
i have 'sudo soup'ed the manager until it reads:

Checking to see if there are hotfixes needed
You are already running the latest version of Security Onion.

i have tried to join a forward node (using SO ISO) to the manager (ubuntu 20.04 network install), as well as a search node (ubuntu 20.04 network install) to the manager....both fail with the same error:

Did not find expectedOutput: 'going to be accepted' in the output below from running the command: 'accept_salt_key_remote'
Attempt to authenticate with the salt master failed with timeout error

i have also tried the install option of "direct" and "direct + manager" and they both fail on both options for the same reasons

please see latest install attempt (forward node SO ISO)
[sosetup.log](https://github.com/Security-Onion-Solutions/securityonion/files/8492086/sosetup.log)

thank you

**Reply by dougburks** [reply]
On the manager, what is the output of `sudo salt-key -L`?

**Comment by 73F0SG1R** [comment]
please see attached
-------- Original Message --------

**Comment by 73F0SG1R** [comment]
![signal-2022-04-19-133309](https://user-images.githubusercontent.com/31665237/164090946-187be7cc-c945-4408-a4a2-d7a63d1b15ae.jpeg)

**Comment by dougburks** [comment]
Looks like your sensor is unable to connect to your manager on the salt ports. Is this related to your other issue at https://github.com/Security-Onion-Solutions/securityonion/discussions/7825?

**Reply by 73F0SG1R** [reply]
no, the same thing happens when trying to add the search node to the manager (distributed arch.) with all ports allowed within AWS VPC.

**Reply by dougburks** [reply]
Is the node able to ssh to the manager with the `soremote` account?

Have you considered using our AMI?
https://docs.securityonion.net/en/2.3/cloud-ami.html

**Reply by 73F0SG1R** [reply]
yes, i can ssh to the manager with soremote

the main reason is that the cloud ami is too expensive. it's just under $4500 / year (per host) for nothing, it doesn't come with support or anything and just because it's in the cloud

**Reply by 73F0SG1R** [reply]
[220421_sosetup.log](https://github.com/Security-Onion-Solutions/securityonion/files/8534434/220421_sosetup.log)
here's the latest sosetup.log failure from this morning

same error, won't accept so manager's salt key

all port/protocols are allowed

**Reply by dougburks** [reply]
> the cloud ami is too expensive. it's just under $4500 / year (per host)

I suspect you are looking at the $0.451/hr price at the top of the page, but that's the TOTAL price and most of that is for the underlying AWS t3a.2xlarge instance. If you're manually installing on your own AWS instance, you're paying those instance fees anyway. The small amount that we charge on top of the instance fees helps to cover our significant AWS costs and to maintain Security Onion itself. Most folks find that it's worth it for the time that they save.
